Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -249,7 +249,7 @@ def self.run_checks(
         # comparison.
         current = if formula
           if formula.head_only?
-            Version.new(formula.any_installed_version.version.commit)
+            Version.new(T.must(T.must(formula.any_installed_version).version.commit))
```

Comment:
I opened https://github.com/Homebrew/brew/pull/19338 to account for these nilable values instead of using `T.must`. It also adds type signatures to related methods in the process, so `brew typecheck` will identify this issue as expected (shout if any of it conflicts with this PR).